crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -279,39 +279,62 @@ pub struct ChildBuilder<'a> {
     push_children: PushChildren,
 }
 
-impl ChildBuilder<'_> {
+/// Trait for building children entities and adding them to a parent entity. See
+/// [`BuildChildren::with_children`].
+pub trait ChildBuild {
+    /// Spawn output type.
```

Comment:
A doc comment on the meaning of the lifetime where clause would really help clarity.